.zuul.yaml

```diff
@@ -0,0 +1,30 @@
+- project:
+    name: kubernetes-sigs/cluster-api-provider-openstack
+    check:
+      jobs:
+        - cluster-api-provider-openstack-conformance-stable
+
+- job:
+    name: cluster-api-provider-openstack-conformance-stable
+    parent: init-test
+    timeout: 36000
+    description: |
+      Test CAPO & Kubernetes master against CAPI & CABPKG stable
+    run: .zuul/playbooks/run.yaml
+    nodeset: ubuntu-xenial-citynetwork
+    secrets:
+      - citynetwork_credentials
```

Comment:
Sorry, It's my fault here.  `secrets` can not be used out of https://github.com/theopenlab/openlab-zuul-jobs repo.

You should do the way like what CPO does: define the job in `openlab-zuul-job` and then use it in CPO

**job defination**: 
https://github.com/theopenlab/openlab-zuul-jobs/blob/master/zuul.d/jobs.yaml#L574
https://github.com/theopenlab/openlab-zuul-jobs/tree/master/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance
**usage**: 
https://github.com/kubernetes/cloud-provider-openstack/blob/master/.zuul.yaml

Reply:
That explains it :)

How is the feedback loop with the opanlab-zuul-jobs repo? I guess I have to get the PR merged there first before I can use it from the cluster-api repo, correct?

So in general it's probably advisable to have the most complicated parts of the test (in my case) in the cluster-api repo to have a comparatively should feedback loop?

Reply:
Yeah. That's the problem. If you use secrets, then all the job defination should be in `openlab-zuul-job` first before cluster-api repo can use. Then it's unable to debug the job in the PR.

Or is the secret must? What will the job do for citynetwork cloud. Maybe we can do it in another way.

Reply:
But calling something like a make target or a shell script in the repo which should be tested (i.e. cluster-api-provider-openstack) should be kind of normal?

This job should create a Kubernetes cluster in Openstack via ClusterAPI and then execute Kubernetes conformance tests against the newly created cluster. So it has to be able to create servers, loadbalancers,.. inside OpenStack

Reply:
`make target` is fine, no matter where the job is defined.

And sorry that citynetwork's loadbalancers server can not be used. CPO lb job is paused for a long time. How about setup a devstack in the job for test?  Openlab provide ansible role to install devstack quickly. https://github.com/theopenlab/openlab-zuul-jobs/tree/master/roles/install-devstack

Reply:
Sound's like the best option when Neutron/Octavia cannot be used directly. I'll take a closer look at devstack.

Reply:
@wangxiyuan Looks like the devstack installation fails. Do I have to use some special nodes or add a secret to use DevStack?